src/peergos/shared/user/UserContext.java

```diff
@@ -1112,16 +1114,16 @@ private FollowRequest decodeFollowRequest(byte[] raw) throws IOException {
         List<EntryPoint> notOurFileSystemEntries = userData.staticData.get()
                 .getEntryPoints(rootKey)
                 .stream()
-                .filter(e -> ! e.owner.equals(ourName))
+                .filter(e -> ! e.ownerName.equals(ourName))
                 .collect(Collectors.toList());
 
         // need to to retrieve all the entry points of our friends
         return Futures.reduceAll(notOurFileSystemEntries, ourRoot, (t, e) -> addEntryPoint(ourName, null, t, e, network, random, fragmenter), (a, b) -> a)
                 .exceptionally(Futures::logError);
     }
 
-    public static EntryPoint convert(String owner, CapabilityWithPath cap) {
-        return new EntryPoint(cap.cap, owner, Collections.emptySet(), Collections.emptySet());
+    public static EntryPoint convert(String ownerName, PublicKeyHash owner, CapabilityWithPath cap) {
+        return new EntryPoint(cap.cap, ownerName, owner, Collections.emptySet(), Collections.emptySet());
```

Comment:
The 2 empty collections doesn't sound right

Reply:
I'm considering getting rid of readers and writers on entry point, because it is information we should calculate from the capability caches.

Reply:
done